### DISCUSSION TITLE
How to change open files limit for the mock build?

### DISCUSSION THREAD

**Discussion by bookwar** [rpm-software-management/mock#1016] [General]
In CentOS Stream CI we run Ansible which runs mock in a `simple` isolation mode to build an rpm package.

The default limit for open files ( 1024) is not enough for some of the larger packages, thus we need to increase it.

It seems that changing ulimit on the mock process doesn't help, as mockbuild user doesn't inherit the limits from the process:
```
[bookwar]$ ulimit -n 10000
[bookwar]$ ulimit -n
10000
[bookwar]$ mock --isolation simple --shell
Start: shell
<mock-chroot> sh-5.2# ulimit -n
10000
<mock-chroot> sh-5.2# su mockbuild
<mock-chroot> bash-5.2$ ulimit -n
1024
```

What is the recommended way to adjust the ulimit in such case?

**Comment by xsuchy** [comment]
Does https://github.com/rpm-software-management/mock/issues/958#issuecomment-1217484186 answer your question?

There is no direct way how to set it in Mock. But we simply call chroot() call. So something like https://superuser.com/questions/381546/limits-conf-ulimits-not-being-applied-to-chrooted-processes would be the way. But I guess the first option I mentioned will be better.
